### ISSUE TITLE
dropColumn doesn't remove foreign key constraint in SQLite

### ISSUE THREAD

**Issue by rijkvanzanten** [knex/knex#4524]
# Environment

Knex version: 0.95.6
Database + version: SQLite 3.31.0
OS: macOS 11.4

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

When using `dropColumn`, any existing foreign key constraints are copied over to the temp table. This causes it to attempt to create a foreign key constraint for a field that isn't included in the create table sql.

2. Error message

```
CREATE TABLE "_knex_temp_alter709" (`id` integer PRIMARY KEY AUTOINCREMENT NOT NULL, `regular_column` varchar(255) NULL, CONSTRAINT related_fk_column_foreign FOREIGN KEY (`fk_column`) REFERENCES `related` (`id`) ON DELETE SET NULL) - SQLITE_ERROR: unknown column "fk_column" in foreign key definition
```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

```js
await knex.schema.createTable('table_a', table => {
  table.increments();
});

await knex.schema.createTable('table_b', table => {
  table.increments();
  table.integer('fk_column').references('id').inTable('table_a');
});

await knex.schema.alterTable('table_b', table => {
  table.dropColumn('fk_column');
});
```

**Comment by joselcvarela** [comment]
I am experiencing this too.

I found no constraints are being removed on drop column:

https://github.com/knex/knex/blob/22002fd8de440fe5e94ce189307d5aa5d6058f41/lib/dialects/sqlite3/schema/ddl.js#L290-L335

The query result of asking the table schema to SQLite is something like this:
```
sqlite> SELECT type, sql FROM sqlite_master WHERE (type="table" OR (type="index" AND sql IS NOT NULL)) AND tbl_name="products_group_content";
type|sql
table|CREATE TABLE "products_group_content" (`id` integer PRIMARY KEY AUTOINCREMENT NOT NULL, `sort` integer NULL, `user_created` char(36) NULL, `date_created` datetime NULL, `user_updated` char(36) NULL, `date_updated` datetime NULL, `name` varchar(255) NOT NULL, `products_group` integer NOT NULL, `template` varchar(255) NOT NULL, `product` integer NULL, `note` integer NULL, `type` varchar(255) not null, `theme` integer null, CONSTRAINT `products_group_content_user_created_foreign` FOREIGN KEY (`user_created`) REFERENCES `directus_users` (`id`), CONSTRAINT `products_group_content_user_updated_foreign` FOREIGN KEY (`user_updated`) REFERENCES `directus_users` (`id`), CONSTRAINT `products_group_content_product_foreign` FOREIGN KEY (`product`) REFERENCES `product` (`id`) ON DELETE SET NULL, CONSTRAINT products_group_content_theme_foreign FOREIGN KEY (`theme`)  REFERENCES `theme` (`id`) ON DELETE SET NULL)
```

But `dropColumn` method don't search in contraints.

**Comment by joselcvarela** [comment]
I have tested at our side and I believe this is now fixed in 0.95.11.
Have tried to remove a foreign column and delete a referenced table and no problems found.

**Comment by kibertoad** [comment]
Released in 0.95.11
